provider/source/src/time_zones/mod.rs

```diff
@@ -291,13 +304,15 @@ impl SourceDataProvider {
                                     .unwrap()
                                     .1;
 
-                                if golden_os.standard != os.standard {
+                                if (!os.is_permanent_dst() && golden_os.standard != os.standard)
+                                    || (os.is_permanent_dst() && golden_os.daylight != os.daylight)
```

Comment:
```suggestion
                                if golden_os.standard != os.standard || (os.standard == os.daylight && golden_os.daylight != os.daylight) {
```

Reply:
that is not the same, `golden_os.standard` and `os.standard` don't match for permanent DST zones, but that's allowed as long as the daylights match